### PR TITLE
Refactor search behaviour

### PR DESCRIPTION
**Pull request by necatikcl**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Bug fix

* **What is the current behavior?** (You can also link to an open issue here)
Whenever `prop.searchText` changes, the component emits `update` with matching results. 

* **What is the new behavior (if this is a feature change)?**
It adds a computed variable `filteredData`, so handling the search doesn't emits update.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
Yes. The emit `update` is renamed to `update:nodes`.

### CODE REVIEW

File: src/lib/components/Tree.vue [hunk 1]
```diff
@@ -53,7 +53,7 @@
 </template>
 
 <script>
-import { watch, reactive, onMounted, computed } from 'vue';
+import { watch, reactive, onMounted, computed, ref } from 'vue';
```
Comment (aykutkardas) [inline src/lib/components/Tree.vue:56]: Is `ref` used?

Comment (necatikcl) [reply src/lib/components/Tree.vue:56]: No, reactive too. Removed 👍

**Comment by alpercun** [comment]
The update event works as expected when the checkbox is clicked and item deletion occurs. Can you update the document to accommodate these changes? @necatikcl

**Review by alpercun — APPROVED** [review]

**Comment by aykutkardas** [comment]
> Yes. The emit update is renamed to update:nodes.

Why? Is there anything in the tree to update other than nodes? @necatikcl

**Comment by necatikcl** [comment]
> > Yes. The emit update is renamed to update:nodes.
> 
> Why? Is there anything in the tree to update other than nodes? @necatikcl

Nope, it is for offering the opportunity to use v-model. `props.nodes` and the data emitted by `update` are equal, so why not use it for v-model?

**Review by denizcdemirci — APPROVED** [review]

**Review by kadiryazici — APPROVED** [review]

**Review by aykutkardas — APPROVED** [review]

**Pull request merged**